src/components/Tabs/Tabs.tsx

```diff
@@ -113,21 +114,50 @@ export const TabBase = ({
 
     const valueIndex = nameToIndexMap[value];
 
-    const animationRef = useRef(new Animated.Value(0));
+    const positionAnimationRef = useRef(new Animated.Value(0));
+    const widthAnimationRef = useRef(new Animated.Value(0));
     const scrollViewRef = useRef<RNScrollView>(null);
     const scrollViewPosition = useRef(0);
 
-    const tabItemPositions = useRef<
-        Array<{ position: number; width: number; contentWidth: number }>
-    >([]);
+    const tabItemPositions = useRef<Array<{ width: number; contentWidth: number }>>([]);
     const [tabContainerWidth, setTabContainerWidth] = useState(0);
 
+    const itemPositionsMap = useMemo(() => {
+        return tabItemPositions.current.reduce((acc, item, index) => {
+            const previousItemsWidth = tabItemPositions.current.reduce((totalWidth, _item, i) => {
+                if (index <= i) return totalWidth;
+
+                totalWidth += _item.width || 0;
+
+                return totalWidth;
+            }, 0);
+
+            acc[index] =
+                variant === 'primary'
+                    ? previousItemsWidth + (item.width - item.contentWidth) / 2
+                    : previousItemsWidth;
+
+            return acc;
+        }, {} as Record<number, number>);
+        // to make useMemo in sync with the ref
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [variant, tabItemPositions.current.length]);
+
+    const itemWidthsMap = useMemo(() => {
+        return tabItemPositions.current.reduce((acc, item, index) => {
+            acc[index] = variant === 'primary' ? item.contentWidth : item.width;
+
+            return acc;
+        }, {} as Record<number, number>);
+        // to make useMemo in sync with the ref
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [variant, tabItemPositions.current.length]);
+
     const scrollHandler = useCallback(
         (currValue: number) => {
             if (tabItemPositions.current.length > currValue) {
-                const itemStartPosition =
-                    currValue === 0 ? 0 : tabItemPositions.current[currValue - 1].position;
-                const itemEndPosition = tabItemPositions.current[currValue].position;
+                const itemStartPosition = currValue === 0 ? 0 : itemPositionsMap[currValue - 1];
```

Comment:
currValue || itemPositionsMap[currValue - 1];

Reply:
Wouldn't this ignore the 0?